edgraph/server.go

```diff
@@ -1573,6 +1613,62 @@ func parseRequest(qc *queryContext) error {
 	return validateQuery(qc.dqlRes.Query)
 }
 
+// addQueryIfUniqueTrue modifies the request if @unique defined on Predicate
+func addQueryIfUniqueTrue(ctx context.Context, qc *queryContext) {
+	if len(qc.gmuList) == 0 {
+		return
+	}
+
+	qc.uniqueVars = map[string]uniqueVarsUids{}
+	for _, gmu := range qc.gmuList {
+
+		if !isUnique(gmu, ctx) {
+			continue
+		} else {
+			buildQuery := ""
+			for i, singlePred := range gmu.Set {
+				predSchema, _ := schema.State().Get(ctx, x.NamespaceAttr(singlePred.Namespace, singlePred.Predicate))
+				if predSchema.Unique {
+					queryVar := fmt.Sprintf("UID%v_", strconv.Itoa(i))
+					//use fmt.SprintF for buils queryma
+					buildQuery = buildQuery + fmt.Sprintf(`%v as var(func: eq(%v,"%v"))
+				{
+					uid
+				}`, queryVar, singlePred.Predicate, singlePred.ObjectValue.GetDefaultVal())
+
+					// So here we are adding query for predicate if unique is defined
+					// for example  Schema => email: string @unique @index(exact) .
+					// To ensure uniqueness of email predicate we will query for value of predicate
```

Comment:
To ensure uniqueness of email predicate, we will query for the value of predicate to ensure that we are not adding a duplicate value.